### PR TITLE
Implement autodelete settings and rework dialog system & settings in general a bit

### LINKED ISSUE #1552: drag'n'dropping multiple files does not send/ask for all files
seems as if only the last is sent.

was just wondering why the recipient dies not get the 3 mp3's i've just sent :)

### PR DESCRIPTION
**Pull request by Jikstra**
- Updated translations
- Reworked dialog system to allow creating dialogs from a passed (lambda) function & we can now spawn multiple dialogs of the same type/class
- Implemented autodelete settings
- Implemented new SettingsButton which looks better in my opinion
- Converted Settings dialog to tsx
- Spltitted out Settings-ManageKeys & Settings-Backup

closes #1552

setting-autodelete
settings-button

### CODE REVIEW

File: scss/dialogs/_settings.scss [hunk 1]
```diff
@@ -77,3 +77,58 @@
...
 }
+
+.SettingsButton {
+  position: relative;
+  height: 40px;
+  width: calc(100% + 40px);
+  left: -20px;
+
+  &:hover {
+    background-color: var(--chatListItemBgHover);
```
[the remark this answers is not in the document]
Comment (Jikstra) [reply scss/dialogs/_settings.scss:88]: I don't like having different hover colors. But I'm fine with renaming that one.

File: scss/dialogs/_settings.scss [hunk 2]
```diff
@@ -77,3 +77,58 @@
...
+.SettingsSelector {
+  position: relative;
+  height: 40px;
+  width: calc(100% + 40px);
+  left: -20px;
+  display: flex;
+  justify-content: space-between;
+
+  &:hover {
+    background-color: var(--chatListItemBgHover);
```
[the remark this answers is not in the document]
Comment (Jikstra) [reply scss/dialogs/_settings.scss:113]: No need to get super granular in my opinion

File: scss/dialogs/_settings.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply scss/dialogs/_settings.scss:88]: for the chatlist we have 3 different backgrounds and that means 3 different hover colors.

File: scss/dialogs/_settings.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply scss/dialogs/_settings.scss:113]: yeah you can use the same as for the one above, just different name than chatlistitem

File: scss/dialogs/_settings.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jikstra) [reply scss/dialogs/_settings.scss:88]: True, but they are on a different background color then white. So to rephrase it, I don't want having different hover colors on top of white ;)

**Comment by r10s** [comment]
nice!

just a comment to the screenshot-animation (btw, also great, for a quick review :)

- the "OK" button should read "OK", not "Save"
- might already be done, however, mentioning it here: there should be an additional confirmation-dialog using the texts `autodel_device_ask` and `autodel_server_ask` with _first placeholder_ replaces by the number of affected messages (via `dc_estimate_deletion_cnt()`) and _second placeholder_ replaced by the selected option string (`autodel_after_1_hour` etc.)

**Comment by Jikstra** [comment]
@r10s thanks!
The confirmation dialog isn't implemented yet, but will do so.

**Comment by Simon-Laux** [comment]
So as discussed with jikstra, I'll take over this pr and rebase it to the new master

**Comment by Jikstra** [comment]
@Simon-Laux lets merge it, the confirmation dialog we can implement in a seperate pr, this thing is already huuuuge

**Pull request merged**